docs/link-data-object.md

```diff
@@ -0,0 +1,455 @@
+        $ref: Group
+    # - rel: subject_Device
+    # - rel: subject_BiologicallyDerivedProduct
+
+```
+
+**Note**:
+
+* JSON document writers must disambiguate polymorphic references.
+* A FHIR specimen's full complement of links: ['additive', 'authorReference_Organization', 'authorReference_Patient', 'authorReference_Practitioner', 'authorReference_PractitionerRole', 'authorReference_RelatedPerson', 'collector_Patient', 'collector_Practitioner', 'collector_PractitionerRole', 'collector_RelatedPerson', 'device', 'location', 'parent', 'procedure', 'request', 'subject_BiologicallyDerivedProduct', 'subject_Device', 'subject_Group', 'subject_Location', 'subject_Patient', 'subject_Substance']
+
+
+```yaml
+specimens:
+- id: s1
+  subject:
+    reference: Patient/1
+- id: s2
+  subject:
+    reference: Group/1
+```
+
+MUST produce:
+
+```yaml
+specimens:
+- id: s1
+  subject:
+    reference: Patient/1
+  links:
+    - rel: subject_Patient
+      href: Patient/1
+- id: s2
+  subject:
+    reference: Group/1
+  links:
+    - rel: subject_Group
+      href: Group/1
+
+```
+
+## The Association Use Case
+
+Association vertices are used for relationships between two objects. They consist of at least two foreign keys, each of which references one of the two objects.
+
+![](association_use_case.jpg)
+
+> The link's relation type identifies its semantics. It is the primary means of conveying how an application can interact with a resource.
+
+In the graph association use case satisfies the following requirements:
+* Defines a typed schema for a given association:
+* The association is a vertex, an edge with data, which the schema author can define an arbitrary set of properties on the edge.
+* When populating the href of an association link, the link writer retrieves the values of template variables from the target vertices.  This differs from [standard hypermedia template processing](https://json-schema.org/draft/2019-09/json-schema-hypermedia.html#rfc.section.7.2.1), where the template variables are retrieved from the instance.
+
+Notes for association use case:
+* No naming constraints apply, the schema author can name `rel` values as they see fit to populate a the `backref` of properties in the target vertex.  Useful for graphql or code generation use cases
+* the `rel` field is used to describe the `backref` of property in the target vertex.
+* the `title` field of the schema is used to describe the `label` of edge between two nodes.
+
+
+```yaml
+"$schema": https://json-schema.org/draft/2020-12/schema
+"$id": https://bmeg.io/foo-bar-association
+title: FooBarAssociation
+properties:
+  data:
+    type: object
+    properties:
+      fizz:
+        type: string
+  links:
+    type: array
+    items:
+      "$ref": https://json-schema.org/draft/2020-12/links
+
+links:
+- rel: bar
+  href: urn:uuid:{id}
+  targetSchema:
+    "$ref": Foo
+  templatePointers:
+  - "/id"
+  targetHints:
+    multiplicity:
+      - has_many
+    directionality:
```

Comment:
This field is referred to as "direction" earlier in the document. Is it different between vertex schemas vs edge schemas, or which is it?

Reply:
Thanks.  I'll correct the discrepancy. It applies to both vertexes and edges